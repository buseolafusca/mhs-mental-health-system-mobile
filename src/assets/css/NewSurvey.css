.NewSurvey-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}
.SurveyResult{
  padding-bottom: 0.5rem;  
}

#page-container {
  position: relative;
  min-height: 100vh;
}

.sv_row{
  text-align: left;
}

.NewSurvey-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.NewSurvey-title {
  font-size: 1.5em;
}

.NewSurvey-intro {
  font-size: large;
}

@keyframes NewSurvey-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
