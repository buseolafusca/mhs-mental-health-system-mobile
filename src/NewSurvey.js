import React, { Component } from "react";
import * as Survey from "survey-react";
import "survey-react/survey.css";
import SurveyCreator from "./SurveyCreator";
import logo from "./logo.svg";
import "./NewSurvey.css";
import "bootstrap/dist/css/bootstrap.css";

import "jquery-ui/themes/base/all.css";

import "nouislider/distribute/nouislider.css";
import "select2/dist/css/select2.css";
import "bootstrap-slider/dist/css/bootstrap-slider.css";

import "jquery-bar-rating/dist/themes/css-stars.css";

import $ from "jquery";
import "jquery-ui/ui/widgets/datepicker.js";
import "select2/dist/js/select2.js";
import "jquery-bar-rating";

import * as widgets from "surveyjs-widgets";
import getQuestionnaire from './BackendService';

import "icheck/skins/square/blue.css";
window["$"] = window["jQuery"] = $;
require("icheck");

Survey.StylesManager.applyTheme("default");

widgets.icheck(Survey, $);
widgets.select2(Survey, $);
widgets.inputmask(Survey);
widgets.jquerybarrating(Survey, $);
widgets.jqueryuidatepicker(Survey, $);
widgets.nouislider(Survey);
widgets.select2tagbox(Survey, $);
widgets.signaturepad(Survey);
widgets.sortablejs(Survey);
widgets.ckeditor(Survey);
widgets.autocomplete(Survey, $);
widgets.bootstrapslider(Survey);

class NewSurvey extends Component {

  json = {
    title: "PHQ-9",
    showProgressBar: "top",
    pages: [
      {
        elements: [
          {
            "type": "radiogroup",
            "name": "Question1",
            "title": "Little interest or pleasure in doing things",
            "isRequired": true,
            "colCount": 0,
            "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
          }
        ]
      },
      {
        questions: [
          {
            "type": "radiogroup",
            "name": "Question2",
            "title": "Feeling down, depressed, or hopeless",
            "isRequired": true,
            "colCount": 0,
            "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
          }
        ]
      },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question3",
      //       "title": "Trouble falling or staying asleep, or sleeping much.",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question4",
      //       "title": "Feeling tired or having little energy",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question5",
      //       "title": "Poor appetite or overeating",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question6",
      //       "title": "Feeling bad about yourself — or that you are a failure or have let yourself or your family down",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question7",
      //       "title": "Trouble concentrating on things, such as reading the newspaper or watching television.",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      // {
      //   questions: [
      //     {
      //       "type": "radiogroup",
      //       "name": "Question8",
      //       "title": "Moving or speaking so slowly that other people could have noticed? Or the opposite — being so fidgety or restless that you have been moving .around a lot more than usual",
      //       "isRequired": true,
      //       "colCount": 0,
      //       "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
      //     }
      //   ]
      // },
      {
        questions: [
          {
            "type": "radiogroup",
            "name": "Question9",
            "title": "Thoughts that you would be better off dead or of hurting yourself in some way",
            "isRequired": true,
            "colCount": 0,
            "choices": ["1|Not At All", "2|Several Days", "3|More the half the days", "4|Nealy"]
          }
        ]
      }
    ],
    completedHtml: "<p><h4>Your Score</h4></p><p>Question 1:<b>{item}</b></p>"
  };

  sendResult(){
    console.log("value changed!");
  }

  onValueChanged(result) {
    console.log("value changed!");
  }

  sendResultOnPageNext() {
    console.log("sendResultOnPageNext");
  }


  goNextPageAutomatic() {
    console.log("goNextPageAutomatic");
  }
  
  onComplete(result) {
    var item = 0;
    console.log("Complete! ");
    console.log(result);
    console.log(result.valuesHash);
    //console.log(result.valuesHash.Question1);
    Object.keys(result.valuesHash).map(function (key) {
      item = item + parseInt(result.valuesHash[key], 10) ;
      console.log(item);
    })
    console.log("Final Result:" + item);

  }


  componentWillMount() {
    console.log("componentWillMount logs");
    const id = "5d0ce7a7fc101609e9765de6";

    getQuestionnaire(id);
  }

  componentDidMount() {
    console.log("componentDidMount logs");
  }

  render() {
    Survey.Survey.cssType = "bootstrap";
    var model = new Survey.Model(this.json);

    return (
      <div className="App">
        <div className="surveyjs">
          {/*If you want to show survey, uncomment the line below*/}
          <Survey.Survey
            model={model}
            onComplete={this.onComplete}
            onValueChanged={this.onValueChanged}

          />
          {/*If you do not want to show Survey Creator, comment the line below*/}
          {/*<h1>SurveyJS Creator in action:</h1>
          <SurveyCreator /> */}
        </div>
       
      </div>
    );
  }
}

export default NewSurvey;
