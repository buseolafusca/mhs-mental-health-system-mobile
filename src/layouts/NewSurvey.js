import React, { Component } from "react";
import * as Survey from "survey-react";
import "survey-react/survey.css";
import logo from "../assets/images/logo.svg";
import "icheck/skins/square/blue.css";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.css";
import "jquery-ui/themes/base/all.css";
import "nouislider/distribute/nouislider.css";
import "select2/dist/css/select2.css";
import "bootstrap-slider/dist/css/bootstrap-slider.css";
import "jquery-bar-rating/dist/themes/css-stars.css";
import $ from "jquery";
import "jquery-ui/ui/widgets/datepicker.js";
import "select2/dist/js/select2.js";
import "jquery-bar-rating";
import * as widgets from "surveyjs-widgets";

import { getQuestionnaire, postAnswers, sendResults } from '../services/BackendService';
import SurveyCreator from "./SurveyCreator";
import "../assets/css/NewSurvey.css";

window["$"] = window["jQuery"] = $;
require("icheck");

Survey.StylesManager.applyTheme("darkblue");

widgets.icheck(Survey, $);
widgets.select2(Survey, $);
widgets.inputmask(Survey);
widgets.jquerybarrating(Survey, $);
widgets.jqueryuidatepicker(Survey, $);
widgets.nouislider(Survey);
widgets.select2tagbox(Survey, $);
widgets.signaturepad(Survey);
widgets.sortablejs(Survey);
widgets.ckeditor(Survey);
widgets.autocomplete(Survey, $);
widgets.bootstrapslider(Survey);

class NewSurvey extends Component {

  constructor(props) {
    super(props);
    this.state = {
      questionnaireId:"5d1a1d16d910160030d04979" ,
      json:
      {
       title: "",
        description: "",
        completedHtml: "",
        pages: [],
        showProgressBar: ""
      }
    };
  }

  /* The following 4 methods are needed by SurveyJS */
  sendResult(){
    console.log("value changed!");
  }

  onValueChanged = (result) => {
    console.log("value changed!");
  }

  sendResultOnPageNext() {
    console.log("sendResultOnPageNext");
  }


  goNextPageAutomatic() {
    console.log("goNextPageAutomatic");
  }


  onComplete = (result) => {
    var finalScore = 0;
    var tableData;
    var i = 1;
    tableData = "<tr><th scope='col'>" + "Question" + "</th><th scope='col'>" + " Answer" + "</th></tr>"
    Object.keys(result.valuesHash).map(function (key) {

      tableData += "<tr>"
      tableData += "<td >" + "Question " + i + "</td>"
      finalScore = finalScore + parseInt(result.valuesHash[key], 10);

      tableData += "<td >" + result.valuesHash[key] + "</td>";
      tableData += "</tr>"
      i++;

    })
    console.log(result);
    console.log(this.model);
    console.log(this.state.questionnaireId);
    console.log(this.model);
    console.log(this.model.data);
    postAnswers(this.model,this.state);
    $("#tbody1").html(tableData);
    document.querySelector('#finalScore').textContent = "Final score is " + finalScore;
    document.querySelector('#jsonSection').textContent = "Result JSON:\n" + JSON.stringify(result.data, null, 3);
  };

  componentWillMount() {
    const { id } = this.props.match.params;
    const url = id;
    this.setState({questionnaireId:id});
    getQuestionnaire(url)
      .then(fetchedData => {
        this.setState({ json: fetchedData.body });
      })
      .catch(error => {
        console.error(error);
      });
  }

  render() {
    Survey.Survey.cssType = "bootstrap";
    this.model = new Survey.Model(this.state.json);
    return (
      <div className="SurveyResult">
        <div className="surveyjs" >
          <Survey.Survey
            model={this.model}
            onComplete={this.onComplete}
            onValueChanged={this.onValueChanged}
          />
          <center>
            <table border="1" width="180" >
              <tbody id="tbody1">

              </tbody>
            </table>
          </center>
          <div id="finalScore"></div>
          <div id="jsonSection"></div>
        </div>

      </div>
    );
    
  }
}

export default NewSurvey;

