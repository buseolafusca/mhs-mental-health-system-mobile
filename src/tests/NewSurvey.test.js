import React from 'react'
import ReactDOM from 'react-dom'
import NewSurvey from '../layouts/NewSurvey'


it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<NewSurvey />, div)
  ReactDOM.unmountComponentAtNode(div)
})
