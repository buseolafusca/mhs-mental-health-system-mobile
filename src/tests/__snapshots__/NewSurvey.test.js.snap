// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSurvey should render correctly in "debug" mode 1`] = `
<div
  className="SurveyResult"
>
  <div
    className="surveyjs"
  >
    <Survey
      model={
        Object {
          "showProgressBar": "",
        }
      }
      onComplete={[Function]}
      onValueChanged={[Function]}
    />
    <center>
      <table
        border="1"
        width="180"
      >
        <tbody
          id="tbody1"
        />
      </table>
    </center>
    <div
      id="finalScore"
    />
    <div
      id="jsonSection"
    />
  </div>
</div>
`;
